### ISSUE TITLE
Frame.to_pandas() does not preserve name of IndexHierarchy

### ISSUE THREAD

**Issue by gdevanla** [InvestmentSystems/static-frame#144]
When converting a Frame with a hierarchical index to Pandas DataFrame, the index `name` is not carried over.

```
ipdb> inner_df
<Frame: inner_df1>
<Index>                              0        1        2        <int64>
<IndexHierarchy: inner_df1>
a                           1        0.1      100      True
a                           2        0.2      200      False
b                           1        0.3      300      True
b                           2        0.4      100      False
<object>                    <object> <object> <object> <object>
ipdb> inner_df.index.name
'inner_df1'
ipdb> inner_df.index.to_pandas().name
ipdb> 
```

Thanks for all the immutability!

**Comment by flexatone** [comment]
Thanks for posting this issue!

This has been fixed, and will be released in 0.5.8.

```python
>>> from static_frame import Frame
>>> f = Frame.from_records(
...             [['a', 1, 10], ['a', 2, 200], ['b', 1, -3], ['b', 2, 7]],
...             columns=('x', 'y', 'z'))
>>> df = f.set_index_hierarchy(['x', 'y']).to_pandas()
>>> df
     x  y    z
x y           
a 1  a  1   10
  2  a  2  200
b 1  b  1   -3
  2  b  2    7
>>> df.index
MultiIndex(levels=[['a', 'b'], [1, 2]],
           labels=[[0, 0, 1, 1], [0, 1, 0, 1]],
           names=['x', 'y'])
```

**Comment by gdevanla** [comment]
I think this is still not fixed in 0.5.9.  Perhaps, something changed?


In [5]: df = pd.DataFrame(data=dict( 
   ...:         a=[0.1, 0.2, 0.3, 0.4], 
   ...:         b=[100, 200, 300, 100], 
   ...:         c=[True, False, True, False])) 
   ...: df.index = pd.MultiIndex.from_tuples([('a', 1), ('a', 2), ('b', 1), ('b', 2)]) 
   ...: df.columns = pd.MultiIndex.from_tuples([('a', 1), ('a', 2), ('b', 1)]) 
   ...: df.index.name = 'Sample_Index' 
   ...: print(df) 
   ...: frame = sf.Frame.from_pandas(df) 
   ...: print(frame) 
   ...:  
   ...:  
   ...:                                                                                                                
       a           b
       1    2      1
a 1  0.1  100   True
  2  0.2  200  False
b 1  0.3  300   True
  2  0.4  100  False
<Frame>
<IndexHierarchy: (None, None)>          a         a       b      <object>
                                        1         2       1      <object>
<IndexHierarchy: (None, None)>
a                              1        0.1       100     True
a                              2        0.2       200     False
b                              1        0.3       300     True
b                              2        0.4       100     False

**Comment by flexatone** [comment]
Note that in Pandas, the `Index.name` attribute has seemingly no connection to `Index.names` attribute. In the example shown above, you are setting `Index.name`. When reading a `pd.MultiIndex` into StaticFrame, we presently only try to transfer `Index.names`, as that is typically where hierarchical index labels are stored. 

It is not clear if we can clearly determine to read from `Index.name` if `Index.names` is not defined. I.e., in this particular case, should we interpret `(None, None)` as a condition by which we look to `name` instead of `names`?

```python
In [8]: df.index.name = 'Sample_Index'                                                                                         

In [9]: df                                                                                                                     
Out[9]: 
       a           b
       1    2      1
a 1  0.1  100   True
  2  0.2  200  False
b 1  0.3  300   True
  2  0.4  100  False

In [10]: df.index.name                                                                                                         
Out[10]: 'Sample_Index'

In [11]: df.index.names                                                                                                        
Out[11]: FrozenList([None, None])
```
